### ISSUE TITLE
Rework permission system & message integrity check

### ISSUE THREAD

**Issue by merlinfuchs** [merlinfuchs/embed-generator#31]
In the current system when a user sends, the bot will check if the user can perform the actions contained in the message. Then it stores a hash of the message together with the message id in the database. Once an interaction is received the bot checks if the hash and message id match a value in the database. If that's not the case the interaction is ignored.

This serves two purposes:
- Users can't perform privilege escalations (e.g. by using a role action to give them a higher role)
- Users can't manually use a webhook produced by the `/webhook` to send arbitrary messages that contain arbitrary actions

This system works for the current set of actions but is kinda hard to extent because every message in an interaction chain has to go through the permission check system. It also requires the computation of the message hash to not change in the future (e.g. if discord changes the data model the hash wouldn't match anymore)

**A more flexible system could look like this:**
When a message is sent the permissions of the member that is sending the message are computed. This also includes the position of the members highest role in the role hierarchy. These two values are then stored in the database together with the message id. Once an interaction is received we can then check if the member that sent the message had the permissions required to perform the action. 
An interaction response would then derive its permissions from the original message. We can just map the interaction response id to the original message id in redis.

This way we can also support actions that dynamically produce other actions. (e.g. custom scripts)

TLDR: a message derives its own permissions from the user when they are sending the message

**Comment by merlinfuchs** [comment]
Instead of storing the permissions and the highest role position we could also store all the role ids that the user had when they sent the message. This way we can compute both values when the interaction is received.

**Comment by merlinfuchs** [comment]
Done with rewrite.
